### ISSUE TITLE
Pathfinder 2 Compatibility

### ISSUE THREAD

**Issue by DariusWolfe** [ironmonk88/TerrainLayer#4]
(recreating this issue from wsaunder's defunct TerrainLayer module)

At current, the terrain icons multiply the movement cost through a tile, which I understand is how difficult terrain works in D&D5E.

I would like to see the option for additive difficulty. For example, PF2E (and possibly PF1E) have two terrain modifiers: Difficult Terrain and Greater Difficult Terrain. Difficult Terrain adds 5 to the movement cost, and Greater Difficult Terrain adds 10 to the movement cost.

Mostly this affects diagonals, because in PF2E, diagonals alternate between costing 5 feet of movement and 10 feet of movement (which averages out to 7.5 feet per tile, close to what it would be with no grid). So on normal terrain, moving 3 tiles diagonally would cost 20 feet of movement. Moving 3 tiles diagonally across difficult terrain would be 30 feet of movement, and 3 tiles across greater difficult terrain would cost 45. With the 2x multiplier, 3 tiles diagonally is 40 feet of movement (5 more than it should be) and 3x is 60 (10 more than it should be)

So to summarize: Add an option for additive (+5/+10, or +x, +2x, etc) difficult terrain.

**Comment by ironmonk88** [comment]
I'm trying to leave static values out of the TerrainLayer and leave it up to the ruler to calculate distances.  TerrainLayer just simply records what type of terrain there is.
So I think this would end up being an issue with whatever ruler you're using.
Currently DragRuler should support PF2E.  In combination with TerrainRuler and TerrainLayer it should solve the issue.

**Comment by DariusWolfe** [comment]
Actually, it looks like the functionality I want has been addressed with
other mods, so there's nothing more to worry about with this request.
Thanks!

**Comment by Ghilteras** [comment]
@DariusWolfe what other mods?

**Comment by DariusWolfe** [comment]
The PF2E Drag Ruler Integration module does a lot of the work. Between Drag Ruler, Enhanced Terrain Layer and the aforementioned integration module, it seems to work smoothly for PF2E.

**Comment by Ghilteras** [comment]
@DariusWolfe @ironmonk88 sorry what do you mean with the aforementioned modules? I am not sure this works as of today for pf2e as this module and Difficult Terrain are not compatible with v9

**Comment by DariusWolfe** [comment]
I have not tested it recently as I haven't noticed any issues to prompt further testing, but at last usage, it seems to work with the current versions of Foundry and PF2E with the modules listed (PF2E Drag Ruler Integration module, Drag Ruler, Enhanced Terrain Layer)

**Comment by ironmonk108** [comment]
Yeah, this module was abandoned awhile back in favor of Enhanced Terrain Layer.
